### PR TITLE
Hardening

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintUiPlugin.java [hunk 1]
```diff
@@ -274,13 +262,9 @@ public static IEditorPart findActiveEditor() {
...
 
-  public static void analyzeCurrentFile() {
+  public static void analyzeOpenedFiles() {
     // SLE-122 Delay a little bit to let the time to the workspace to initialize (and avoid NPE)
-    new AnalyzeCurrentFileJob().schedule(2000);
-  }
-
-  public static void addSonarLintPartListener() {
-    new RegisterSonarLintPartListenerJob().schedule();
+    new AnalyzeOpenedFilesJob().schedule(2000);
```
Comment (dbmeneses) [inline org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintUiPlugin.java]: I guess there is no callback to know when the workspace is ready?

File: org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/bind/ProjectBindModel.java [hunk 2]
```diff
@@ -95,6 +97,9 @@ public String getServerId() {
   public void setAutoBindFailed(boolean autoBindFailed) {
     String oldValue = getSonarFullName();
     this.autoBindFailed = autoBindFailed;
+    this.moduleKey = null;
+    this.serverId = null;
+    this.server = null;
```
Comment (dbmeneses) [inline org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/bind/ProjectBindModel.java]: maybe extract this to a method, like `String oldValue = resetBinding()` ?

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/server/Server.java [hunk 3]
```diff
@@ -307,7 +309,14 @@ public void stop() {
 
   @Override
   public synchronized void updateStorage(IProgressMonitor monitor) {
-    updateStatus = client.update(getConfig(), new WrappedProgressMonitor(monitor, "Update configuration from server '" + getId() + "'")).status();
+    UpdateResult updateResult = client.update(getConfig(), new WrappedProgressMonitor(monitor, "Update configuration from server '" + getId() + "'"));
+    for (SonarAnalyzer analyzer : updateResult.analyzers()) {
+      if (!analyzer.sonarlintCompatible() && analyzer.minimumVersion() != null) {
```
Comment (dbmeneses) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/server/Server.java]: This doesn't seem correct. Check [SLI](https://github.com/SonarSource/sonarlint-intellij/blob/master/src/main/java/org/sonarlint/intellij/tasks/ServerUpdateTask.java#L95)

Comment (henryju) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/server/Server.java]: My bad, I misunderstood the sonarlintCompatible() (I though it was already taking into account the minimumVersion)

File: org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintUiPlugin.java (hunk 1 above)
Comment (henryju) [inline org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintUiPlugin.java]: Everything is lazy, so I'm not sure it means anything "workspace ready". In any case, I don't know any event that is triggered when all editors are restored.

**Pull request merged**